src/overlays/actors/ovl_Bg_Spot17_Bakudankabe/z_bg_spot17_bakudankabe.c

```diff
@@ -21,13 +20,145 @@ const ActorInit Bg_Spot17_Bakudankabe_InitVars = {
+        } else {
+            gravityInfluence = -420;
+        }
+
+        if (Math_Rand_ZeroOne() < 0.4f) {
+            rotationSpeed = 65;
+        } else {
+            rotationSpeed = 33;
+        }
+        func_80029E8C(globalCtx, &burstDepthY, &burstDepthX, &burstDepthY, gravityInfluence, rotationSpeed, 0x1E, 4, 0,
+                      scale, 1, 3, 80, -1, OBJECT_GAMEPLAY_FIELD_KEEP, D_0500A880);
+    }
+    Math_Vec3f_Copy(&burstDepthY, &this->dyna.actor.posRot.pos);
+    func_80033480(globalCtx, &burstDepthY, 60.0f, 4, 110, 160, 1);
+    burstDepthY.y += 40.0f;
+    func_80033480(globalCtx, &burstDepthY, 60.0f, 4, 120, 160, 1);
+    burstDepthY.y += 40.0f;
+    func_80033480(globalCtx, &burstDepthY, 60.0f, 4, 110, 160, 1);
+}
+
+void BgSpot17Bakudankabe_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot17Bakudankabe* this = THIS;
+    s32 pad;
+    s32 sp24 = 0;
+
+    DynaPolyInfo_SetActorMove(&this->dyna.actor, 0);
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params & 0x3F))) {
+        Actor_Kill(&this->dyna.actor);
+        return;
+    }
+    DynaPolyInfo_Alloc(&D_06000A38, &sp24);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, sp24);
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+}
+
+void BgSpot17Bakudankabe_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot17Bakudankabe* this = THIS;
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void BgSpot17Bakudankabe_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot17Bakudankabe* this = THIS;
+    if (this->dyna.actor.xzDistFromLink < 650.0f && func_80033684(globalCtx, &this->dyna.actor) != NULL) {
+        func_808B6BC0(this, globalCtx);
+        Flags_SetSwitch(globalCtx, (this->dyna.actor.params & 0x3F));
+        Audio_PlaySoundAtPosition(globalCtx, &this->dyna.actor.posRot.pos, 40, NA_SE_EV_WALL_BROKEN);
+        func_80078884(NA_SE_SY_CORRECT_CHIME);
+        Actor_Kill(&this->dyna.actor);
+    }
+}
+
+void BgSpot17Bakudankabe_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    {
+        u32 gameplayFrames;
+        s8 sp62;
+        s8 sp63;
+        GraphicsContext* gfxCtx;
+        Gfx* dispRefs[4];
+
+        sp62 = coss((u16)(globalCtx->gameplayFrames * 1500)) >> 8;
+        sp63 = coss((u16)(globalCtx->gameplayFrames * 1500)) >> 8;
+
+        gfxCtx = globalCtx->state.gfxCtx;
+        Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_bg_spot17_bakudankabe.c", 269);
+
+        func_80093D18(globalCtx->state.gfxCtx);
+
+        gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_spot17_bakudankabe.c", 273),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+        sp62 = (sp62 >> 1) + 0xC0;
+        sp63 = (sp63 >> 1) + 0xC0;
+        gameplayFrames = globalCtx->gameplayFrames;
+        gDPSetEnvColor(gfxCtx->polyOpa.p++, sp62, sp63, 0xFF, 0x80);
```

Comment:
Color values should be decimal (as of recently)
```suggestion
        gDPSetEnvColor(gfxCtx->polyOpa.p++, sp62, sp63, 255, 128);
```